pkg/tool/osdctl/osdctl.go

```
@@ -0,0 +1,199 @@
+	if strings.TrimSpace(binarySum) != strings.TrimSpace(checksum) {
+		fmt.Printf("WARNING: Checksum for osdctl does not match the calculated value. Please retry installation. If issue persists, this tool can be downloaded manually at %s\n", osdctlBinaryAsset.GetBrowserDownloadURL())
+		// We shouldn't link this binary to latest if the checksum isn't valid
+		return nil
+	}
+
+	if err := unArchive(t.toolDir(""), filepath.Join(versionedDir, checksumAsset.GetName())); err != nil {
+		return fmt.Errorf("failed to unarchive the checksum file '%s': %w", checksumFilePath, err)
+	}
+	//Untar osdctl file and remove (so, it removes .tar.gz basically)
+
+	// Link as latest
+	latestFilePath := t.symlinkPath(latestDir)
+	err = os.Remove(latestFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to remove existing 'osdctl' binary at '%s': %w", latestDir, err)
+	}
+
+	err = os.Symlink(osdctlBinaryFilepath, latestFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to link new 'osdctl' binary to '%s': %w", latestDir, err)
+	}
+	return nil
+}
+
+// toolDir returns this tool's specific directory given the root directory all tools are installed in
+func (t *Tool) toolDir(rootDir string) string {
+	return filepath.Join(rootDir, "osdctl")
+}
+
+func (t *Tool) symlinkPath(latestDir string) string {
+	return filepath.Join(latestDir, "osdctl")
+}
+
+func (t *Tool) Remove(rootDir, latestDir string) error {
+	// Remove all binaries owned by this tool
+	toolDir := t.toolDir(rootDir)
+	err := os.RemoveAll(toolDir)
+	if err != nil {
+		return fmt.Errorf("failed to remove %s: %w", toolDir, err)
+	}
+
+	// Remove all symlinks owned by this tool
+	latestFilePath := t.symlinkPath(latestDir)
+	err = os.Remove(latestFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to remove symlinked file %s: %w", latestFilePath, err)
+	}
+	return nil
+}
+
+func unArchive(source string, destination string) error {
+	src, err := os.Open(source)
+	if err != nil {
+		return nil
```

Lets return the error encountered, formatted with context (`fmt.Errorf("failed to open tarball '%s': %v",source, err)` for example)

---

Thanks Trevor, Let me update that